### PR TITLE
I have a working decoder

### LINKED ISSUE #79: Create an endpoint that will take in an invitation URL and receive it through the didcomm service

### LINKED ISSUE #80: Add frontend input for submitting an invitation URL

### PR DESCRIPTION
**Pull request by Vanderscycle**
working on issue #79 and #80 (frontend and backend)

### CODE REVIEW

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<script lang="ts">
```
Comment (brianorwhatever) [inline frontend/src/lib/slideOverComponents/submitCredentials.svelte]: The name of this file doesn't make sense. It is invitation accepting not credential submitting

File: backend/src/presentations/presentations.service.ts [hunk 2]
```diff
@@ -128,4 +128,10 @@ export class PresentationsService {
       ...updatePresentationRequest,
     });
   }
+
+  async acceptInvitation(url: any): Promise<any> {
```
Comment (brianorwhatever) [inline backend/src/presentations/presentations.service.ts]: is this function used? if not lets remove it

File: backend/src/presentations/presentations.service.ts [hunk 3]
```diff
@@ -128,4 +128,10 @@ export class PresentationsService {
...
+    const regrex = /(?<==).*/gmi
+    let sha64 = regrex.exec(url)
```
Comment (brianorwhatever) [inline backend/src/presentations/presentations.service.ts]: whats a sha64? 😺

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  }
+};
+//NOTE:weird implementation but allows for both 'enter' and the sumbit button to work
+async function acceptInvitationApiCall(url: string) {
+  if (urlInput) {
+    let response = await acceptInvitation({ url: urlInput });
+    if (response) {
+      swal({
+        title: "Success",
+        text: `You have submitted the encrypted data for processing`,
```
Comment (brianorwhatever) [inline frontend/src/lib/slideOverComponents/submitCredentials.svelte]:
```suggestion
        text: `You have accepted the invitation`,
```

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        icon: "success",
+        button: "Great!",
+      }).then(slideOverContent.set(null));
+    } else {
+      swal({
+        title: "something went wrong",
+        text: "please ensure you have pasted the url from the QrCode in the input box. Otherwise, please contact us for it may be on our end.",
```
Comment (brianorwhatever) [inline frontend/src/lib/slideOverComponents/submitCredentials.svelte]:
```suggestion
        text: "Please ensure you have submitted a valid Invitation URL.",
```

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    if (response) {
+      swal({
+        title: "Success",
+        text: `You have submitted the encrypted data for processing`,
+        icon: "success",
+        button: "Great!",
+      }).then(slideOverContent.set(null));
+    } else {
+      swal({
+        title: "something went wrong",
```
Comment (brianorwhatever) [inline frontend/src/lib/slideOverComponents/submitCredentials.svelte]:
```suggestion
        title: "Error",
```

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      });
+    }
+
+    //error
+  }
+}
+</script>
+
+<template>
+  <h1>Hello there!, you will have to submit the url here</h1>
```
Comment (brianorwhatever) [inline frontend/src/lib/slideOverComponents/submitCredentials.svelte]:
```suggestion
  <h1>Enter a valid Invitation URL</h1>
```

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  <Button
+    label="submit"
+    callback="{() => {
+      acceptInvitationApiCall(urlInput);
+    }}" />
+  <input
+    bind:value="{urlInput}"
+    on:keypress="{onKeyPress}"
+    type="text"
+    placeholder="paste the url and press enter" />
```
Comment (brianorwhatever) [inline frontend/src/lib/slideOverComponents/submitCredentials.svelte]:
```suggestion
    placeholder="Enter a valid Invitation URL" />
```

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte (hunk 7 above)
Comment (Vanderscycle) [reply frontend/src/lib/slideOverComponents/submitCredentials.svelte]: oh yeah, It was the initial component placeholder that I should have removed. Modified it accordingly

File: backend/src/presentations/presentations.service.ts (hunk 2 above)
Comment (Vanderscycle) [reply backend/src/presentations/presentations.service.ts]: Make sense, I do confuse both structures sometimes. fixed by pr 399e500f80978d4aef2a96b4b021a953ee426776

File: backend/src/presentations/presentations.service.ts (hunk 3 above)
Comment (Vanderscycle) [reply backend/src/presentations/presentations.service.ts]: deleted the useless block of code in pr 399e500f80978d4aef2a96b4b021a953ee426776

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte (hunk 4 above)
Comment (Vanderscycle) [reply frontend/src/lib/slideOverComponents/submitCredentials.svelte]: fixed by pr 399e500f80978d4aef2a96b4b021a953ee426776

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte (hunk 5 above)
Comment (Vanderscycle) [reply frontend/src/lib/slideOverComponents/submitCredentials.svelte]: fixed by pr 399e500f80978d4aef2a96b4b021a953ee426776

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte (hunk 6 above)
Comment (Vanderscycle) [reply frontend/src/lib/slideOverComponents/submitCredentials.svelte]: fixed by pr 399e500f80978d4aef2a96b4b021a953ee426776

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte (hunk 7 above)
Comment (Vanderscycle) [reply frontend/src/lib/slideOverComponents/submitCredentials.svelte]: fixed by pr 399e500f80978d4aef2a96b4b021a953ee426776

File: frontend/src/lib/slideOverComponents/submitCredentials.svelte (hunk 8 above)
Comment (Vanderscycle) [reply frontend/src/lib/slideOverComponents/submitCredentials.svelte]: fixed by pr 399e500f80978d4aef2a96b4b021a953ee426776

**Pull request merged**